analyzer/analyzer.spicy

```diff
@@ -2,6 +2,9 @@
 
 module DHCP;
 
+# Global variable containing the DHCP message type of the currently parsed message
+public global dhcp_type: MessageType;
+
```

Comment:
I have the feeling that with such a global where we do not tie passed options with a message anymore it will become much easier to e.g., incorrectly use a type passed with a previous message. I'd suggest to instead get rid of the global and extract any passed type in `create_msg` (basically a `message_type`-specific piece of code like we have for other options in `create_options`), e.g.,

```ruby
    local message_type: DHCP::MessageType;
    for (option in msg.options.options) {
        if (option?.message_type)
            message_type = option.message_type;
    }

    return (
        cast<uint32>(msg.op),
        cast<uint32>(message_type),
        ...
```

Reply:
Done (even added a break to the loop, since here we only look for one option)